### ISSUE TITLE
JXCore roadmap

### ISSUE THREAD

**Issue by TimNZ** [jxcore/jxcore#153]
Hi,

Can you please be transparent around your roadmap for JXCore - without open sourcing or making code available, and providing visibility of where JXCore is heading you will throw away a lot of the good work as people will be hesitant to commit to production use.

**Comment by TimNZ** [comment]
What's the latest update?

**Comment by obastemur** [comment]
Right this moment; Implemented;

- Multiple JS engines in place and both passes the core test cases (including multithreading support) and stress tests.
- Embedders API is implemented. (Anyone interested may embed jxcore with (one of the JS engines) into their apps (for any platform)
- UI wrapper around embedded interface works on Android, and iOS 
- MemoryWrapper, and ThreadWrapper modules are improved. 

Long story short we are almost there. 
- Debugger support for the second JS engine (seems like leaving this for after release)
- Performance / stability updates on Secondary engine and iOS target (working on it)
- Documentations / tests.. (working on it)

**Comment by TimNZ** [comment]
Whoa. 

I am looking forward to getting my hands on this, but I re-iterate my last comment - you are piling a lot of time and effort into what seems like the "ultimate" V1, I don't want most of that, it is a bunch of distinct/separate functionality.

**Comment by TimNZ** [comment]
I say this as someone who was a classic tech startup founder, building first, finding out if anyone wanted it second.

**Comment by ktrzeciaknubisa** [comment]
Tim, I completely agree with your point of view - doing things, about which you're not sure if they will find any user for themselves. I might have done this exact mistake few times in my personal life before.

This is different however. Android and iOS users are waiting to be able to run JXcore on their devices. Those and others are also waiting for open sourcing. Yes, we have jumped into big water to deliver it all at once. But we also enjoy swimming :)

**Comment by sashahilton00** [comment]
+1 from me for open sourcing it soon ;) Looking forward to finally getting multithreaded nodejs, which is truly cross platform compatible :) When this is eventually open sourced, I have a feeling that there is going to be a big uptqake in v. little time at all ;)

**Comment by cref** [comment]
Not sure if this is the right place to ask but anyway:
Where does JXCore stand on supporting ES6?
Or, to put it differently: wouldn't it be a better idea to 'fork' [io.js](https://iojs.org/) instead of Node with its conservative approach?

**Comment by obastemur** [comment]
You are all welcome to participate in a related discussion #176 
This one served the purpose, and now closing it. 

@cref our release gave you the answers I suppose, feel free to continue from the other discussion. 

Thank you all!
